auction-server/src/opportunity/service/get_quote.rs

```diff
@@ -68,6 +69,31 @@ use {
 // while keeping the original amount (before fees) in the bid
 // --------------------------------------------------------------------------------------------
 
+/// Prefix for indicative price taker keys
+/// The first 28 bytes of "Price11111111111111111111111111111111111112"
+pub const INDICATIVE_PRICE_TAKER_BASE: [u8; 28] = [
+    0x05, 0xda, 0xfe, 0x58, 0xfc, 0xc9, 0x54, 0xbe, 0x96, 0xc9, 0x32, 0xae, 0x8e, 0x9a, 0x17, 0x68,
+    0x9d, 0x10, 0x17, 0xf8, 0xc9, 0xe1, 0xb0, 0x7c, 0x86, 0x32, 0x71, 0xc0,
+];
+
+/// For users that don't provide a wallet we assign them a random public key prefixed by INDICATIVE_PRICE_TAKER_BASE
+pub fn generate_indicative_price_taker() -> Pubkey {
+    let mut key_bytes = [0u8; 32];
+
+    key_bytes[0..28].copy_from_slice(&INDICATIVE_PRICE_TAKER_BASE);
+
+    let mut rng = rand::thread_rng();
+    let rand_bytes: [u8; 4] = rng.gen(); // Generate all four bytes at once
```

Comment:
any reason to limit this to just 4 bytes of randomness? why not 8 and totally future proof it?